const asyncWrap = (fn) => {
  return async (req, res, next) => {
    try {
      await fn(req, res, next);
    } catch (error) {
      next(error);
    }
  };
};
// asyncwrap middleware to handle async errors
module.exports = asyncWrap;
